src/utils/getUrlByRequestOptions.ts

```diff
@@ -19,8 +19,8 @@ export function getUrlByRequestOptions(
   if (!options.protocol) {
     debug('given no protocol, resolving...')
 
-    // Assume HTTPS if using an SSL certificate.
-    options.protocol = options.cert ? 'https:' : DEFAULT_PROTOCOL
+    // Assume HTTPS if uri is https or if using an SSL certificate.
+    options.protocol = options.uri?.protocol || (options.cert ? 'https:' : DEFAULT_PROTOCOL)
```

Comment:
I'm afraid this may be incorrect. If a request has been issued with a certificated attached, we must treat it as an HTTPS request, regardless of what protocol the URL may have. `options.cert` should take the higher priority. What do you think about reversing the addition:

```js
options.protocol = options.cert ? 'https:' : options.uri?.protocol || DEFAULT_PROTOCOL
```

Reply:
That's probably better. Although, this is the case where you try to do a `https` request to an URL starting with `http://`? I think Node already throws an error in that case.

Reply:
There are request issuing libraries that perform an HTTPS request without explicitly providing the `protocol: 'https:'` option. Instead, they have the `cert` property set. This condition is designed for that kind of libraries.

Overall, there are so many ways to perform an HTTP request in node using the `http` module it's mesmerizing. Some of them won't even compile against the type definition of the functions they call, yet they are present.